code/game/objects/items/trader.dm

```diff
@@ -148,3 +148,502 @@
+	w_class = W_CLASS_TINY
+	icon_state = "vamphead0"
+	flags = HEAR | FPRINT
+	var/obj/effect/decal/cleanable/blood/located_blood
+	var/last_used = 0
+
+/obj/item/device/vampirehead/New()
+	..()
+	processing_objects += src
+
+/obj/item/device/vampirehead/Destroy()
+	processing_objects -= src
+	..()
+
+/obj/item/device/vampirehead/process()
+	if(located_blood && get_dist(located_blood,src)<5)
+		return //Don't process further, we still smell our old blood
+	if(ishuman(loc))
+		var/mob/living/carbon/human/H = loc
+		if(genecheck(H,prob(97))) //Annoy unqualified bearers with messages about 3% of the time
+			find_blood()
+	located_blood = null
+	update_icon()
+
+/obj/item/device/vampirehead/dropped()
+	..()
+	located_blood = null
+
+/obj/item/device/vampirehead/update_icon()
+	icon_state = "vamphead[located_blood ? "1" : "0"]"
+
+/obj/item/device/vampirehead/proc/find_blood()
+
+
+	update_icon()
+
+/obj/item/device/vampirehead/on_enter_storage(obj/item/weapon/storage/S)
+	..()
+	var/mob/living/carbon/human/H = get_holder_of_type(src, /mob/living/carbon/human)
+	if(H && genecheck(H,TRUE)) //If we've been stashed by a valid user. Don't send normal reject messages.
+		var/list/reject_phrases = list("Don't put me in there, I can't see!",
+								"Miserable cur! Un[S] me at once!",
+								"I hope you realize the view inside here is terribly boring.",
+								"What do you want of me? To curate the inside of this [S]?",
+								"Miserable. Sealed inside a [S].",
+								"Can't make it! Can't make it! [capitalize(S)] stuck! Please, I beg you!",
+								"This is really no substitute from a coffin.",
+								"What, pray tell, am I supposed to be doing inside here?")
+		to_chat(H,"<B>[src]</B> [pick("murmurs","shrieks","hisses","groans","complains")], \"<span class='sinister'>[pick(reject_phrases)]</span>\"")
+
+/obj/item/device/vampirehead/proc/genecheck(mob/user,silent=FALSE)
+	if(!ishuman(user))
+
+		return FALSE
+	if(M_SOBER in user.mutations)
+		return TRUE
+	else
+		if(!silent)
+			var/list/reject_phrases = list("Blah! I'd never work with someone who can't hold their drink.",
+										"You are not meant for this line of work, featherweight.",
+										"Try again when you can relate to the intoxicating taste of blood.",
+										"What is a man? A miserable little pile of soft drinks.",
+										"You mortals all look underage to me. Pray tell, can you even manage a bottle?",
+										"Bah. You mock Le Confr�rie des Chevaliers du Tastevin with your plebian visage.",
+										"I will not associate with any less than an iron liver.",
+										"You dare ask my service when you cannot even hold your liquor?")
+			to_chat(user,"<B>[src]</B> [pick("murmurs","insults","mocks","groans","complains")], \"<span class='sinister'>[pick(reject_phrases)]</span>\"")
+		return FALSE
+
+/obj/item/device/vampirehead/attack(mob/living/M as mob, mob/user as mob)
```

Comment:
You don't appear to be using last_used anywhere. Was it supposed to be used as a cooldown for this flash attack?

Reply:
Yes it was, good catch